src/PaymentManager/PaymentManager.Services/Implementations/PayableService.cs

```diff
@@ -78,5 +79,54 @@ public async Task<PayableRead> UpdateAsync(int agencyId, int id, PayableUpdate m
             var updatedEntity = await _repository.UpdateAsync(entity);
             return _mapper.Map<PayableRead>(updatedEntity);
         }
+
+        public async Task UpdateSupplier(int agencyId, int id, PayableSupplierUpdate model)
+        {
+            var entityQuery = await _repository.Find(m => m.AgencyId == agencyId && m.SupplierId == model.SupplierId);
+            var payables = entityQuery.ToList();
+
+            foreach (var payable in entityQuery)
+            {
+                payable.SupplierName = model.SupplierName;
+            }
+
+            await _repository.UpdateRange(payables);
+        }
+
+        public async Task UpdatePayablesTourStatus(int agencyId, int tourId, int tourStatus)
+        {
+            var payables = await _repository.Find(m => m.AgencyId == agencyId && m.TourId == tourId);
+            var model = payables.ToList();
+            foreach (var payable in model)
+            {
+                payable.TourStatus = (TourStatus)tourStatus;
+            }
+
+            await _repository.UpdateRange(model);
+        }
+
+        public async Task DeleteSupplierFromPayable(int agencyId, int tourId, int supplierId)
+        {
+            var payables = await this.Find(m => m.AgencyId == agencyId && m.TourId == tourId && m.SupplierId == supplierId);
+            if (payables.Count != 1)
+            {
+                throw new Exception(); //????
+            }
+
+            await this.Remove(agencyId, payables[0].Id);
+        }
+
+        public async Task Remove(int agencyId, int id)
+        {
+            var entity = await _repository.GetById(agencyId, id);

```

Comment:
Add `Async` suffix and everywhere where missed.
Also _check failures from GitHub Actions_
![image](https://user-images.githubusercontent.com/15414389/149498264-8cbc8ac0-05bf-4fd1-80c9-370a1f21104c.png)